include/juicyc/preprocessor_impl.cc

```diff
@@ -0,0 +1,72 @@
+//Created by FENG, H
+#include "preprocessor_impl.h"
+#include <sstream>
+#include <string>
+#include <fstream>
+#include <map>
+#include <regex>
+using namespace std;
+namespace juicy {
+
+	void Preprocessor_impl::process_define()
+	{
+		size_t pos;
+		map <string, string> macro_map;
+		ifstream in;
+		string lineStr, segment, tmp = "", key, value;
+		for(auto it: input_files_names)
+		{
+			tmp = "";
+			in.open(it,ios::in);
+			if(!in)
+			{
+				cout << "File does not exist!" << endl;
```

Comment:
Use juicyc::Status for error handling. e.g. return `Status::IOError("File does not exist")` here